Assets/Scripts/WorldGenTestScripts/WorldGenerator.cs

```diff
@@ -2,75 +2,182 @@
+        }
+    }
+
+    private static Pixel ChooseConnection(List<Pixel> connectionsX, List<Pixel> connectionsY, bool horizontalConnecting)
+    {
+        return horizontalConnecting ? connectionsX[random.Next(connectionsX.Count)] :
+            connectionsY[random.Next(connectionsY.Count)];
+    }
+
+    private static bool chooseConnectionOrientation(List<Pixel> connectionsX, List<Pixel> connectionsY)
+    {
+        return connectionsX.Count * connectionsY.Count != 0 ? (random.Next(2) % 2 == 1) : connectionsY.Count != 0 ? false : true;
+    }
+
+    private static bool anyConnections(List<Pixel> connectionsX, List<Pixel> connectionsY)
+    {
+        return connectionsX.Count != 0 || connectionsY.Count != 0;
+    }
+
+    private void GenerateTiles()
+    {
+        for (int i = 0; i < width; i++)
+        {
+            for (int j = 0; j < height; j++)
+            {
+                pixels[i, j].GenerateTile(transform);
+            }
+        }
+    }
+
+    void CreateRoomBorders() 
+    {
+        for (int i = 0; i < width; i++)
+        {
+            for (int j = 0; j < height; j++)
+            {
+                if (IsAreaOccupied(new Vector2(i, j), new Vector2(i, j), 1) &&
+                    pixels[i, j].GetState() == State.EMPTY &&
+                    !OnWorldBorder(new Vector2(i,j)))
+                {
+                    ChangePixelColor(new Vector2(i,j), State.BORDER);
+                }
+            }
+        }
+    }
+
+    void RemoveDeadEnds()
+    {
+        for (int i = 0; i < width; i++)
+        {
+            for (int j = 0; j < height; j++)
             {
-                ChangePixelColor(new Vector2((float)x, (float)y), 2);
+                if ((pixels[i, j].GetState() == State.CORIDOR_X || pixels[i, j].GetState() == State.CORIDOR_Y)
+                    && IsEntranceToNowhere(i,j))
+                {
+                    ChangePixelColor(new Vector2((float)i, (float)j), State.EMPTY);
+                }
             }
         }
-        Debug.Log("location: " + point1.x + "," + point1.y + ";" + point2.x + "," + point2.y);
     }
 
-    bool IsOccupied(Vector2 point)
+    private bool IsEntranceToNowhere(int x, int y)
     {
-        return ( point.x<0 || point.y<0 || point.x >= width || point.y >= height || pixels[(int)point.x, (int)point.y].colorId == 2);
+        return !(IsOccupied(new Vector2(Mathf.Max(x - 1, 0),y)) && IsOccupied(new Vector2(Mathf.Min(x + 1, width - 1), y))) &&
+               !(IsOccupied(new Vector2(x,Mathf.Max(y - 1, 0))) && IsOccupied(new Vector2(x,Mathf.Min(y + 1, width - 1))));
     }
 
-    bool IsAreaOccupied(Vector2 point1, Vector2 point2)
+    public bool IsOccupied(Vector2 point)
     {
-        for (int x = (int)point1.x-5; x <= (int)point2.x+5; x++)
+        return ( OutsideWorldBorder(point) ||
+            (pixels[(int)point.x, (int)point.y].GetState() != State.EMPTY &&
+            pixels[(int)point.x, (int)point.y].GetState() != State.BORDER));
```

Comment:
multiple calls to GetState
```suggestion
        State state = pixels[(int)point.x, (int)point.y].GetState();
        return OutsideWorldBorder(point) ||
            (state != State.EMPTY && state != State.BORDER);
```